pkg/vm/create_evm.go

```diff
@@ -103,6 +106,14 @@ func CreateEVMGenesis(
 		params.initialTokenAllocation[common.HexToAddress(teleporterInfo.FundedAddress)] = core.GenesisAccount{
 			Balance: balance,
 		}
+		if addICMMessengerToGenesis || addICMRegistryToGenesis {
```

Comment:
logic is not clear for me here. Can we deal with messenger and registry separate. so we have 2 if cases where we check each and deploy each